doc/tutorial/broadcasting.txt

```diff
@@ -0,0 +1,70 @@
+.. _tutbroadcasting:
+
+
+============
+Broadcasting
+============
+
+Broadcasting is a mechanism which allows tensors with
+different numbers of dimensions to be added or multiplied
+together by (virtually) replicating the smaller tensor along
+the dimensions that it is lacking.
+
```

Comment:
Let's explain the general concept of broadcasting here first, before describing how it is implemented in Theano, like it was done in the original documentation (the one in basic.txt, not in sandbox/).
The one in sandbox was not that clear, that's why we wanted to remove it, but maybe we can work some of its content into the new tutorial.

For that, I think the easiest is to include the whole content of the original documentation, and after that, we can expand by adding the Theano code corresponding to the figure (with the same values if possible), then the code to add a column.

We can also add some of the information from sandbox/broadcasting.txt, but it is too verbose, and somehow redundant with the other one.